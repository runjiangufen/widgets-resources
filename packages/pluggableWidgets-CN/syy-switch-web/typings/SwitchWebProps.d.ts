/**
 * This file was generated from SwitchWeb.xml
 * WARNING: All changes made to this file will be overwritten
 * @author Mendix UI Content Team
 */
import { ActionValue, DynamicValue, EditableValue } from "mendix";

export type SizeEnum = "default" | "small";

export type ConfirmTypeEnum = "pop" | "modal";

export type ConfirmokTypeEnum = "primary" | "ghost" | "dashed" | "link" | "text" | "default";

export interface SwitchWebContainerProps {
    name: string;
    tabIndex?: number;
    id: string;
    checked: EditableValue<boolean>;
    onName: string;
    offName: string;
    loadingFlag?: EditableValue<boolean>;
    disabled?: EditableValue<boolean>;
    size: SizeEnum;
    autoFocus: boolean;
    onChange?: ActionValue;
    openconfirm: boolean;
    confirmType: ConfirmTypeEnum;
    ontitle?: DynamicValue<string>;
    offtitle?: DynamicValue<string>;
    confirmokText: string;
    confirmcancelText: string;
    confirmokType: ConfirmokTypeEnum;
    onConfirm?: ActionValue;
    onCancel?: ActionValue;
}

export interface SwitchWebPreviewProps {
    readOnly: boolean;
    checked: string;
    onName: string;
    offName: string;
    loadingFlag: string;
    disabled: string;
    size: SizeEnum;
    autoFocus: boolean;
    onChange: {} | null;
    openconfirm: boolean;
    confirmType: ConfirmTypeEnum;
    ontitle: string;
    offtitle: string;
    confirmokText: string;
    confirmcancelText: string;
    confirmokType: ConfirmokTypeEnum;
    onConfirm: {} | null;
    onCancel: {} | null;
}
