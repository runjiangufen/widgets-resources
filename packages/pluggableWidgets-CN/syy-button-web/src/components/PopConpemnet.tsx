import { ReactNode, ReactElement, createElement, MouseEvent } from "react";
import { Popconfirm } from "antd";
import { ConfirmokTypeEnum } from "../../typings/ButtonWebProps";

export interface PopConpemnetProps {
    confirmtitle?: string;
    confirmokText?: string;
    confirmcancelText?: string;
    confirmokType?: ConfirmokTypeEnum | undefined;
    onConfirm?: (event: MouseEvent<HTMLDivElement>) => void;
    onCancel?: (event: MouseEvent<HTMLDivElement>) => void;
    onClickAction?: (event: MouseEvent<HTMLDivElement>) => void;
    children: ReactNode;
}

// confirmation pop-up
export function PopConpemnet(props: PopConpemnetProps): ReactElement {
    return (
        <Popconfirm
            style={{ padding: 0, backgroundColor: "red" }}
            title={props.confirmtitle}
            okText={props.confirmokText}
            okType={props.confirmokType}
            cancelText={props.confirmcancelText}
            onConfirm={props.onConfirm}
            onCancel={props.onCancel}
        >
            {props.children}
        </Popconfirm>
    );
}
