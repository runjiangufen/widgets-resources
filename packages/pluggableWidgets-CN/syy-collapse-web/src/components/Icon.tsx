import { createElement, ReactElement } from "react";
import classNames from "classnames";
import { WebIcon } from "mendix";

interface IconProps {
    animate?: boolean;
    empty?: boolean;
    full?: boolean;
    value: WebIcon;
}

export function Icon({ animate, empty, full, value }: IconProps): ReactElement {
    if (value && value.type === "glyph") {
        return (
            <span
                className={classNames(
                    "rating-icon",
                    { "rating-icon-empty": empty, "rating-icon-full": full, animate },
                    "glyphicon",
                    value.iconClass
                )}
                aria-hidden="true"
            />
        );
    }
    if (value && value.type === "image") {
        return (
            <img
                className={classNames("rating-image", {
                    "rating-image-empty": empty,
                    "rating-image-full": full,
                    animate
                })}
                src={value.iconUrl}
                alt=""
            />
        );
    }
    return <div />;
}
