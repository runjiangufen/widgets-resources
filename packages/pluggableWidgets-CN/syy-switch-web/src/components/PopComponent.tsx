import { ReactNode, ReactElement, createElement, MouseEvent } from "react";
import Popconfirm from "antd/lib/popconfirm";
import { ConfirmokTypeEnum } from "../../typings/SwitchWebProps";

export interface PopComponentProps {
    confirmtitle?: string;
    confirmokText?: string;
    confirmcancelText?: string;
    confirmokType?: ConfirmokTypeEnum | undefined;
    onConfirm?: (event: MouseEvent<HTMLDivElement>) => void;
    onCancel?: (event: MouseEvent<HTMLDivElement>) => void;
    onClickAction?: (event: MouseEvent<HTMLDivElement>) => void;
    children?: ReactNode;
}

// confirmation pop-up
export function PopComponent(props: PopComponentProps): ReactElement {
    return (
        <Popconfirm
            style={{ padding: 0, backgroundColor: "red" }}
            title={props.confirmtitle}
            okText={props.confirmokText}
            okType={props.confirmokType}
            cancelText={props.confirmcancelText}
            onConfirm={props.onConfirm}
            onCancel={props.onCancel}
        >
            {props.children}
        </Popconfirm>
    );
}
