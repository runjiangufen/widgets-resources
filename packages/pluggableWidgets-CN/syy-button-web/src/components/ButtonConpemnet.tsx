import { ReactElement, createElement } from "react";

import { Button, ButtonProps } from "antd";

import { Icon } from "./Icon";
import { WebIcon } from "mendix";

export type ConfirmokTypeEnum = "primary" | "ghost" | "dashed" | "link" | "text" | "default";

export interface ButtonConpemnetProps extends ButtonProps {
    icon: WebIcon;
}

// ButtonConpemnet
export function ButtonConpemnet(props: ButtonConpemnetProps): ReactElement {
    const { block, href, ghost, loading, danger, disabled, size, icon, shape, type, children, onClick } = props;
    return (
        <Button
            disabled={disabled}
            danger={danger}
            loading={loading}
            ghost={ghost}
            block={block}
            href={href ? href : undefined}
            size={size}
            shape={shape}
            type={type}
            onClick={onClick}
        >
            {icon ? <Icon data={icon} empty /> : ""}
            {children}
        </Button>
    );
}
