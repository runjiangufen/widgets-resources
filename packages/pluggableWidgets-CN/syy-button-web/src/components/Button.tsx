import { ReactElement, createElement, ReactNode } from "react";
import { PopConpemnet, PopConpemnetProps } from "./PopConpemnet";
import { ButtonConpemnet, ButtonConpemnetProps } from "./ButtonConpemnet";

export interface ButtonWebProps extends ButtonConpemnetProps, PopConpemnetProps {
    children: ReactNode;
    openconfirm?: boolean;
    confirmType?: "pop" | "modal";
}
// ButtonConpemnet
export function Button(props: ButtonWebProps): ReactElement {
    const {
        block,
        href,
        ghost,
        loading,
        danger,
        disabled,
        size,
        icon,
        shape,
        type,
        children,
        openconfirm, // Open the confirmation pop-up window before executing microflow
        confirmType,
        confirmtitle,
        confirmokText,
        onClick,
        onConfirm,
        onCancel,
        confirmcancelText,
        confirmokType
    } = props;

    const ButtonWebView = (
        <ButtonConpemnet
            disabled={disabled}
            danger={danger}
            loading={loading}
            ghost={ghost}
            block={block}
            href={href}
            size={size}
            shape={shape}
            type={type}
            onClick={onClick}
            icon={icon}
        >
            {children}
        </ButtonConpemnet>
    );
    return (
        <div>
            {openconfirm && confirmType === "pop" ? (
                <PopConpemnet
                    confirmtitle={confirmtitle}
                    confirmokText={confirmokText}
                    confirmcancelText={confirmcancelText}
                    confirmokType={confirmokType}
                    onConfirm={onConfirm}
                    onCancel={onCancel}
                >
                    {ButtonWebView}
                </PopConpemnet>
            ) : (
                ButtonWebView
            )}
        </div>
    );
}
