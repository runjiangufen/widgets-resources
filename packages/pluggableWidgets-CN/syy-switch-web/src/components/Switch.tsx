import { ReactElement, createElement } from "react";
import { PopComponent, PopComponentProps } from "./PopComponent";
import { SwitchComponent, SwitchComponentProps } from "./SwitchComponent";

export interface SwitchWebProps extends SwitchComponentProps, PopComponentProps {
    openconfirm?: boolean;
    confirmType?: "pop" | "modal";
    offtitle?: string;
    ontitle?: string;
}

export function Switch(props: SwitchWebProps): ReactElement {
    const {
        checked,
        loadingFlag,
        disabled,
        onName,
        offName,
        autoFocus,
        size,
        onClick,
        onChange,
        onConfirm,
        onCancel,
        openconfirm,
        confirmType,
        confirmcancelText,
        confirmokText,
        confirmokType,
        offtitle,
        ontitle
    } = props;
    const SwitchWebView = (
        <SwitchComponent
            checked={checked}
            loadingFlag={loadingFlag}
            disabled={disabled}
            onName={onName}
            offName={offName}
            autoFocus={autoFocus}
            size={size}
            onClick={onClick}
            onChange={onChange}
        />
    );
    const confirmtitle = (checked && offtitle) || ontitle;
    return (
        <div>
            {openconfirm && confirmType === "pop" ? (
                <PopComponent
                    confirmcancelText={confirmcancelText}
                    confirmokText={confirmokText}
                    confirmokType={confirmokType}
                    confirmtitle={confirmtitle}
                    onConfirm={onConfirm}
                    onCancel={onCancel}
                >
                    {SwitchWebView}
                </PopComponent>
            ) : (
                SwitchWebView
            )}
        </div>
    );
}
