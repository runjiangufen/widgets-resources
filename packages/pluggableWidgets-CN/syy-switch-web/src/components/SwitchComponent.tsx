import { ReactElement, createElement } from "react";
import Switch, { SwitchProps } from "antd/lib/switch";

export interface SwitchComponentProps extends SwitchProps {
    loadingFlag: boolean;
    onName: string;
    offName: string;
}

export function SwitchComponent(props: SwitchComponentProps): ReactElement {
    const { checked, loadingFlag, disabled, onName, offName, autoFocus, size, onClick, onChange } = props;
    const options = {
        ...(onName && { checkedChildren: onName }),
        ...(offName && { unCheckedChildren: offName })
    };
    return (
        <Switch
            checked={checked || false}
            autoFocus={autoFocus || false}
            onChange={onChange}
            onClick={onClick}
            disabled={disabled}
            loading={loadingFlag}
            size={size}
            {...options}
        />
    );
}
