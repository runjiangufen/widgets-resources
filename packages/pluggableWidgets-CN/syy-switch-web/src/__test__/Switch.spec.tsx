import { createElement } from "react";
import { render, mount } from "enzyme";
// import { SwitchWeb } from "../SwitchWeb";
import { Switch, SwitchWebProps } from "../components/Switch";
import { SwitchComponent } from "../components/SwitchComponent";
// import { PopComponent, PopComponentProps } from "../components/PopComponent";
// import { EditableValueBuilder, actionValue } from "@mendix/piw-utils-internal";
import { ConfirmTypeEnum, ConfirmokTypeEnum } from "../../typings/SwitchWebProps";

interface SwitchWebContainerPropsMore {
    openconfirm: boolean;
    confirmType?: ConfirmTypeEnum;
    confirmtitle?: string;
    confirmokText?: string;
    confirmcancelText?: string;
    confirmokType?: ConfirmokTypeEnum;
    onConfirm?: () => void;
    onCancel?: () => void;
}

describe("SwitchWeb", () => {
    let defaultSwitchProps: SwitchWebProps;
    let popConfirmSwitchProps: SwitchWebContainerPropsMore;
    beforeEach(() => {
        defaultSwitchProps = {
            checked: false,
            openconfirm: false,
            onName: "",
            offName: "",
            size: "default",
            loadingFlag: false,
            autoFocus: false,
            onClick: jest.fn()
        };
        popConfirmSwitchProps = {
            openconfirm: true,
            confirmType: "pop",
            confirmtitle: "Are you sure to delete this task?",
            confirmokText: "Yes",
            confirmcancelText: "No",
            confirmokType: "primary",
            onConfirm: jest.fn(),
            onCancel: jest.fn()
        };
    });

    it("render default switch with false ", () => {
        const switchWrapper = render(<SwitchComponent {...defaultSwitchProps} />);
        expect(switchWrapper).toMatchSnapshot();
    });

    it("render default switch with true", () => {
        const switchWrapper = render(<SwitchComponent {...defaultSwitchProps} checked />);
        expect(switchWrapper).toMatchSnapshot();
    });

    it("render switch disable", () => {
        const switchWrapper = render(<SwitchComponent {...defaultSwitchProps} disabled />);
        expect(switchWrapper).toMatchSnapshot();
    });

    it("when checked is true and onName is setted renders with correct attributes", () => {
        const switchWrapper = render(<SwitchComponent {...defaultSwitchProps} checked onName="On" />);
        // expect(switchButton.find(".ant-switch-inner").text()).toBe("On");
        expect(switchWrapper).toMatchSnapshot();
    });

    it("when checked is false and offName is setted renders with correct attributes", () => {
        const switchWrapper = render(<SwitchComponent {...defaultSwitchProps} offName="Off" />);
        // expect(switchButton.find(".ant-switch-inner").text()).toBe("Off");
        expect(switchWrapper).toMatchSnapshot();
    });

    it("when openconfirm and type is pop renders with correct attributes", () => {
        const switchWrapper = render(<Switch {...defaultSwitchProps} {...popConfirmSwitchProps} />);
        expect(switchWrapper).toMatchSnapshot();
    });

    it("when openconfirm and type is pop renders with correct attributes and title is show", () => {
        const switchWrapper = render(
            <Switch {...defaultSwitchProps} {...popConfirmSwitchProps} confirmtitle="Hello" />
        );
        expect(switchWrapper).toMatchSnapshot();
    });

    it("when openconfirm and type is modal renders with correct attributes", () => {
        const switchWrapper = render(<Switch {...defaultSwitchProps} {...popConfirmSwitchProps} confirmType="modal" />);
        expect(switchWrapper).toMatchSnapshot();
    });

    it("triggers onClick function with a click event", () => {
        const onClick = jest.fn();
        const switchWrapper = mount(<SwitchComponent {...defaultSwitchProps} onClick={onClick} />);
        const button = switchWrapper.find("button");
        button.simulate("click");
        expect(onClick).toHaveBeenCalledTimes(1);
    });

    it("when checked is true renders with correct attributes", () => {
        const switchWrapper = render(<SwitchComponent {...defaultSwitchProps} checked />);
        expect(switchWrapper.hasClass("ant-switch-checked")).toBe(true);
    });

    it("when checked is false renders with correct attributes", () => {
        const switchWrapper = render(<SwitchComponent {...defaultSwitchProps} />);
        expect(switchWrapper.hasClass("ant-switch-checked")).toBe(false);
    });

    it("when switch is disabled renders with correct attributes", () => {
        const switchWrapper = render(<SwitchComponent {...defaultSwitchProps} checked disabled />);
        expect(switchWrapper.hasClass("ant-switch-disabled")).toBe(true);
    });

    it("when checked is true and onName is ON renders with correct attributes", () => {
        const switchWrapper = render(<SwitchComponent {...defaultSwitchProps} checked onName="ON" />);
        expect(switchWrapper.hasClass("ant-switch-checked")).toBe(true);
        expect(switchWrapper.find("span.ant-switch-inner").text()).toBe("ON");
    });

    it("when checked is false and offName is OFF renders with correct attributes", () => {
        const switchWrapper = render(<SwitchComponent {...defaultSwitchProps} offName="OFF" />);
        expect(switchWrapper.hasClass("ant-switch-checked")).toBe(false);
        expect(switchWrapper.find("span.ant-switch-inner").text()).toBe("OFF");
    });
});

// interface SwitchWebContainerPropsMore extends SwitchWebContainerProps {
//     onClick: () => void;
// }
// describe("SwitchWeb", () => {
//     let switchWrapper: ReactWrapper<SwitchWebContainerProps, any>;
//     let switchComponent: ReactWrapper<SwitchWebProps, any>;
//     let switchButtonComponent: ReactWrapper<SwitchComponentProps, any>;
//     let switchButtonWrapper: ReactWrapper<any, any>;
//     let switchButton: ReactWrapper<any, any>;
//     let pop: ReactWrapper<PopComponentProps, any>;
//     // let modal: ReactWrapper<any, any>;
//     const createAndFindElements = (props: SwitchWebContainerPropsMore): void => {
//         switchWrapper = mount(<SwitchWeb {...props} />);
//         switchComponent = switchWrapper.find(Switch);
//         switchButtonComponent = switchComponent.find(SwitchComponent);
//         switchButtonWrapper = switchButtonComponent.find("Wave").find("Switch");
//         switchButton = switchButtonWrapper.find(".ant-switch");
//         pop = switchComponent.find(PopComponent);
//         // switchButton = switchWrapper.find(Switch);
//         // pop = switchWrapper.find(Popconfirm);
//         // modal = switchWrapper.find(Modal);
//     };
//     const createProps = (props?: Partial<SwitchWebContainerPropsMore>): SwitchWebContainerPropsMore => {
//         const defaultProps: SwitchWebContainerPropsMore = {
//             name: "sYYSwitchWeb",
//             tabIndex: 0,
//             id: "runjian.switchweb.SwitchWeb1",
//             checked: new EditableValueBuilder<boolean>().withValue(false).build(),
//             onName: "",
//             offName: "",
//             size: "default",
//             autoFocus: false,
//             openconfirm: false,
//             confirmType: "pop",
//             confirmokText: "ok",
//             confirmcancelText: "cancel",
//             confirmokType: "primary",
//             onConfirm: undefined,
//             onCancel: undefined,
//             onClick: jest.fn()
//         };

//         return { ...defaultProps, ...props };
//     };

//     it("with editable value renders the structure correctly", () => {
//         createAndFindElements(
//             createProps({
//                 checked: new EditableValueBuilder<boolean>().withValue(false).build()
//             })
//         );

//         expect(switchWrapper).toMatchSnapshot();
//     });

//     it("when openconfirm and type is pop renders with correct attributes", () => {
//         const props = { openconfirm: true, onConfirm: actionValue(), onCancel: actionValue() };
//         createAndFindElements(createProps({ ...props }));
//         switchButton.simulate("click");
//         expect(props.onConfirm.execute).toHaveBeenCalledTimes(1);
//         expect(switchWrapper).toMatchSnapshot();
//     });

//     it("when openconfirm and type is modal renders with correct attributes", () => {
//         const onClick = jest.fn();
//         const Wrapper = mount(<Switch offName="" onName="" loadingFlag={false} onClick={onClick} />);
//         const button = Wrapper.find("button");
//         button.simulate("click");
//         expect(onClick).toHaveBeenCalledTimes(1);
//         // createAndFindElements(createProps({ openconfirm: true, confirmType: "modal" }));
//         // switchButton.simulate("click");
//         // expect(switchWrapper.hasClass("ant-modal-root")).toBe(true);
//         expect(switchWrapper).toMatchSnapshot();
//     });

//     it("when openconfirm and click onComfirm renders with correct attributes", () => {
//         createAndFindElements(createProps({ openconfirm: true }));
//         switchButton.simulate("click");
//         // pop.find("Tooltip").props().onClick("onConfirm");
//         expect(pop).toMatchSnapshot();
//     });

//     it("with tabIndex passed renders correctly", () => {
//         createAndFindElements(createProps({ tabIndex: 1 }));

//         expect(switchWrapper.props().tabIndex).toEqual(1);
//     });

//     // it("without tabIndex passed renders correctly", () => {
//     //     createAndFindElements(createProps({ tabIndex: undefined }));

//     //     expect(switchWrapper.props().tabIndex).toEqual(0);
//     // });

//     describe("when disable", () => {
//         it("renders elements with correct attributes", () => {
//             const props = createProps({
//                 disabled: new EditableValueBuilder<boolean>().withValue(true).build()
//             });

//             createAndFindElements(props);

//             expect(switchButton.hasClass("ant-switch-disabled")).toBe(true);
//         });

//         it("shouldn't change the attributes value", () => {
//             const props = createProps({
//                 disabled: new EditableValueBuilder<boolean>().withValue(true).build()
//             });

//             createAndFindElements(props);
//             switchButton.simulate("click");

//             expect(props.checked.setValue).not.toHaveBeenCalled();
//         });

//         it("shouldn't invoke action", () => {
//             const props = createProps({
//                 disabled: new EditableValueBuilder<boolean>().withValue(true).build(),
//                 onChange: actionValue()
//             });
//             createAndFindElements(props);

//             switchWrapper.simulate("click");

//             expect(props.onChange?.execute).not.toHaveBeenCalled();
//         });
//     });
// });
