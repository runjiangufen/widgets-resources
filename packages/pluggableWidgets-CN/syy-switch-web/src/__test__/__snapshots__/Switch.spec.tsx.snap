// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwitchWeb render default switch with false  1`] = `
<button
  aria-checked="false"
  class="ant-switch"
  role="switch"
  type="button"
>
  <div
    class="ant-switch-handle"
  />
  <span
    class="ant-switch-inner"
  />
</button>
`;

exports[`SwitchWeb render default switch with true 1`] = `
<button
  aria-checked="true"
  class="ant-switch ant-switch-checked"
  role="switch"
  type="button"
>
  <div
    class="ant-switch-handle"
  />
  <span
    class="ant-switch-inner"
  />
</button>
`;

exports[`SwitchWeb render switch disable 1`] = `
<button
  aria-checked="false"
  class="ant-switch ant-switch-disabled"
  disabled=""
  role="switch"
  type="button"
>
  <div
    class="ant-switch-handle"
  />
  <span
    class="ant-switch-inner"
  />
</button>
`;

exports[`SwitchWeb when checked is false and offName is setted renders with correct attributes 1`] = `
<button
  aria-checked="false"
  class="ant-switch"
  role="switch"
  type="button"
>
  <div
    class="ant-switch-handle"
  />
  <span
    class="ant-switch-inner"
  >
    Off
  </span>
</button>
`;

exports[`SwitchWeb when checked is true and onName is setted renders with correct attributes 1`] = `
<button
  aria-checked="true"
  class="ant-switch ant-switch-checked"
  role="switch"
  type="button"
>
  <div
    class="ant-switch-handle"
  />
  <span
    class="ant-switch-inner"
  >
    On
  </span>
</button>
`;

exports[`SwitchWeb when openconfirm and type is modal renders with correct attributes 1`] = `
<div>
  <button
    aria-checked="false"
    class="ant-switch"
    role="switch"
    type="button"
  >
    <div
      class="ant-switch-handle"
    />
    <span
      class="ant-switch-inner"
    />
  </button>
</div>
`;

exports[`SwitchWeb when openconfirm and type is pop renders with correct attributes 1`] = `
<div>
  <button
    aria-checked="false"
    class="ant-switch"
    role="switch"
    type="button"
  >
    <div
      class="ant-switch-handle"
    />
    <span
      class="ant-switch-inner"
    />
  </button>
</div>
`;

exports[`SwitchWeb when openconfirm and type is pop renders with correct attributes and title is show 1`] = `
<div>
  <button
    aria-checked="false"
    class="ant-switch"
    role="switch"
    type="button"
  >
    <div
      class="ant-switch-handle"
    />
    <span
      class="ant-switch-inner"
    />
  </button>
</div>
`;
